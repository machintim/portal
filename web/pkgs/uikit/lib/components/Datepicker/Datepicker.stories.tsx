import type { Meta, StoryObj } from '@storybook/react';
import { useArgs } from '@storybook/preview-api';
import Datepicker from './Datepicker';

const meta = {
  title: 'Components/Work in progress/Datepicker/Datepicker',
  component: Datepicker,
  parameters: { layout: 'padded' },
  argTypes: {
    date: { control: { type: 'date' } },
    minDate: { control: { type: 'date' } },
    maxDate: { control: { type: 'date' } },
    native: { control: { type: 'boolean' } },
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Datepicker>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    date: null,
    onDateChange: () => {},
  },
  render: () => {
    const [args, updateArgs] = useArgs();

    // Convert UNIX timestamp from Storybook controls to `Date` object
    // https://storybook.js.org/docs/essentials/controls#annotation
    args.date = args.date ? new Date(args.date) : null;
    args.minDate = args.minDate ? new Date(args.minDate) : null;
    args.maxDate = args.maxDate ? new Date(args.maxDate) : null;

    // const onChange = (event: React.ChangeEvent<HTMLInputElement>) => updateArgs({ date: event.target.valueAsDate });
    const onDateChange = (date: Date | null | undefined) =>
      updateArgs({ date: date });
    return (
      <Datepicker
        {...args}
        date={args.date}
        onDateChange={onDateChange}
      />
    );
  },
};

export const AsNative: Story = {
  name: 'As native fallback',
  args: {
    native: true,
    date: null,
    onDateChange: () => {},
  },
  render: () => {
    const [args, updateArgs] = useArgs();
    args.date = args.date ? new Date(args.date) : null;
    args.minDate = args.minDate ? new Date(args.minDate) : null;
    args.maxDate = args.maxDate ? new Date(args.maxDate) : null;
    const onDateChange = (date: Date | null | undefined) =>
      updateArgs({ date: date });
    return (
      <Datepicker
        {...args}
        date={args.date}
        onDateChange={onDateChange}
      />
    );
  },
};
