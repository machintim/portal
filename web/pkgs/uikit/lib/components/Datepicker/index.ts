export { default } from './Datepicker';
export type { DatepickerProps } from './Datepicker';
