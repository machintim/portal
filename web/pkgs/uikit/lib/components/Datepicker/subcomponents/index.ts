export { default as CalendarGrid } from './CalendarGrid';
export { default as CalendarMenu } from './CalendarMenu';
export { default as ColumnHeadings } from './ColumnHeadings';
export { default as DateField } from './DateField';
export { default as Day } from './Day';
export { default as Native } from './Native';
export { default as MonthSelector } from './MonthSelector';
