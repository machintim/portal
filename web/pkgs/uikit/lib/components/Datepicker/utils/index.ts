export { default as getDatesForCalendarGrid } from './getDatesForCalendarGrid/getDatesForCalendarGrid';
export { default as parseDate } from './parseDateForDateField/parseDateForDateField';
