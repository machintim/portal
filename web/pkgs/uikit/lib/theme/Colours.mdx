import { Meta, Title, Subtitle, ColorPalette, ColorItem } from '@storybook/addon-docs/blocks';
import theme from './defaultTheme';

<Meta title="theme/Colours" />
<Title>Colours explorer</Title>
<Subtitle>Colours approved for use in the UCL Design System</Subtitle> 

More information on the use of colour in the UCL Design System <a href="https://www.figma.com/design/8Sm5PxWOWJYpYXAzhRzUzt/UCL-Design-System-UI-Kit?node-id=1-287" target="_blank">can be found on Figma</a>. 

The colours populating this page are imported from the <a href="https://github.com/ucl-isd/uikit-react/blob/main/lib/theme/defaultTheme.ts#L5" target="_blank">theme file</a>. 

Below are all colours per category, one category per line: 

<ColorPalette>
{
  Object.entries(theme.color).map(([key, value]) => {
    return (
      <ColorItem key={key} title={key.charAt(0).toUpperCase() + key.slice(1)} colors={value} />
    );
  })
}
</ColorPalette>

Below is a breakdown of each colour category. These are the same colours as above, but grouped by category, one colour per line: 

{Object.entries(theme.color).map(([key, value]) => {
  return (
    <div key={key}>
      <h2>{key.charAt(0).toUpperCase() + key.slice(1)}</h2>

      <ColorPalette>

      {Object.entries(value).map(([colorKey, colorValue]) => {
        return (
          <ColorItem key={colorKey} title={colorKey} colors={{ [colorKey]: colorValue }} />
        );
      })}

      {console.log(key, value)}

      </ColorPalette>
    </div>
  );
})}
