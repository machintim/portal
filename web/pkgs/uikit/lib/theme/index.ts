export { default as theme } from './defaultTheme';
export type {
  ThemeType,
  DisplayColoursType,
  BlackAndWhiteColoursType,
} from './defaultTheme';
export {
  default as useTheme,
  ThemeContextProvider,
} from './useTheme';
