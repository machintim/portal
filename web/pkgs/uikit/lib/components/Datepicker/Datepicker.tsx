import {
  useState,
  useRef,
  memo,
  InputHTMLAttributes,
} from 'react';
import { css, cx } from '@emotion/css';
import {
  Native,
  DateField,
  CalendarMenu,
} from './subcomponents';
import { useTheme } from '../../theme';

export const NAME = 'ucl-datepicker';

export interface DatepickerProps
  extends InputHTMLAttributes<HTMLInputElement> {
  date: Date | null | undefined;
  minDate?: Date | null | undefined;
  maxDate?: Date | null | undefined;
  onDateChange: (date: Date | null | undefined) => void;
  native?: boolean;
  testId?: string;
}

const Datepicker = ({
  date,
  minDate,
  maxDate,
  onDateChange,
  native,
  testId = NAME,
  className,
  ...props
}: DatepickerProps) => {
  const [theme] = useTheme();
  const [isOpen, setIsOpen] = useState(false);
  const datepickerRef = useRef<HTMLDivElement>(null);

  if (native)
    return (
      <Native
        date={date}
        minDate={minDate}
        maxDate={maxDate}
        onDateChange={onDateChange}
        testId={testId}
        {...props}
      />
    );

  const blur = () =>
    (document.activeElement as HTMLElement)?.blur();

  const handleKeyDown = (
    event: React.KeyboardEvent<HTMLDivElement>
  ) => {
    // Keyboard mappings for interacting with the date `<CalendarMenu>`
    // Escape & Enter both close the `<CalendarMenu>`
    if (event.key === 'Escape' || event.key === 'Enter')
      blur();
    // Move the currently selected date in the `<CalendarGrid />`
    else if (event.key === 'ArrowLeft' && date)
      onDateChange(
        new Date(
          date.getFullYear(),
          date.getMonth(),
          date.getDate() - 1
        )
      );
    else if (event.key === 'ArrowRight' && date)
      onDateChange(
        new Date(
          date.getFullYear(),
          date.getMonth(),
          date.getDate() + 1
        )
      );
    else if (event.key === 'ArrowUp' && date)
      onDateChange(
        new Date(
          date.getFullYear(),
          date.getMonth(),
          date.getDate() - 7
        )
      );
    else if (event.key === 'ArrowDown' && date)
      onDateChange(
        new Date(
          date.getFullYear(),
          date.getMonth(),
          date.getDate() + 7
        )
      );
  };

  const handleOnFocus = () => setIsOpen(true);

  const handleDatePick = () => blur();

  const handleBlurCapture = (event: React.FocusEvent) => {
    // Only close if focus moves outside `<Datepicker>` and its subcomponents
    if (
      !datepickerRef.current?.contains(
        event.relatedTarget as Node
      )
    )
      setIsOpen(false);
  };

  const baseStyle = css`
    position: relative;
    width: max-content;
    user-select: none;

    &:focus-visible,
    &:focus-within {
      box-shadow: ${theme.boxShadow.focus};
      outline: none;
    }
  `;

  const style = cx(baseStyle, className);

  return (
    <div
      className={style}
      ref={datepickerRef}
      onFocus={handleOnFocus}
      onBlurCapture={handleBlurCapture}
      onKeyDown={handleKeyDown}
      tabIndex={0}
      data-testid={testId}
    >
      <DateField
        date={date}
        onChange={onDateChange}
        testId={testId}
      />
      {isOpen && (
        <CalendarMenu
          date={date}
          setDate={onDateChange}
          onDatePick={handleDatePick}
          testId={testId}
        />
      )}
    </div>
  );
};

export default memo(Datepicker);
