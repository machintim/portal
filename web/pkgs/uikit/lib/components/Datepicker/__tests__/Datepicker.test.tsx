import {
  describe,
  test,
  expect,
  vi,
  afterEach,
  beforeAll,
  afterAll,
} from 'vitest';
import {
  render,
  screen,
  fireEvent,
  cleanup,
} from '@testing-library/react';
import { ThemeContextProvider } from '../../../theme';
import Datepicker from '../Datepicker';

const customRender = (ui: React.ReactElement) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeContextProvider>
        {children}
      </ThemeContextProvider>
    ),
  });
};

describe('Datepicker', () => {
  beforeAll(() => {
    // Snapshot tests will fail because the Calendar Grid includes styling for "today's date".
    // Therefore, we need to use Vitest to mock the current date.
    vi.useFakeTimers();
    vi.setSystemTime(new Date('2025-03-10')); // Arbitrary fixed date -- Alex's birthday :)
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  afterEach(() => {
    cleanup();
    vi.resetAllMocks();
  });

  test('Snapshot: no date provided', () => {
    const renderResult = customRender(
      <Datepicker
        date={null}
        onDateChange={vi.fn()}
      />
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('Snapshot: date provided', () => {
    const renderResult = customRender(
      <Datepicker
        date={new Date('2025-02-04')}
        onDateChange={vi.fn()}
      />
    );
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('Snapshot: with :focus & date provided', () => {
    const testId = 'abc';
    const renderResult = customRender(
      <Datepicker
        date={new Date('2025-02-04')}
        onDateChange={vi.fn()}
        testId={testId}
      />
    );
    const dateField = screen.getByTestId(
      testId + '-date-field'
    );
    fireEvent.focus(dateField);
    expect(
      renderResult.container.firstChild
    ).toMatchSnapshot();
  });

  test('Renders without crashing', () => {
    const testId = 'abc';
    customRender(
      <Datepicker
        date={null}
        onDateChange={vi.fn()}
        testId={testId}
      />
    );
    const datepicker = screen.getByTestId(testId);
    expect(datepicker).toBeInTheDocument();
  });

  test('Calendar Menu appears on :focus', () => {
    const testId = 'abc';
    customRender(
      <Datepicker
        date={null}
        onDateChange={vi.fn()}
        testId={testId}
      />
    );
    let calendarMenu = screen.queryByTestId(
      testId + '-calendar-menu'
    );
    expect(calendarMenu).not.toBeInTheDocument();
    const dateField = screen.getByTestId(testId);
    fireEvent.focus(dateField);
    calendarMenu = screen.getByTestId(
      testId + '-calendar-menu'
    );
    expect(calendarMenu).toBeInTheDocument();
  });

  test('Calendar Menu disappears on :blur', () => {
    const testId = 'abc';
    customRender(
      <Datepicker
        date={null}
        onDateChange={vi.fn()}
        testId={testId}
      />
    );
    const dateField = screen.getByTestId(
      testId + '-date-field'
    );
    fireEvent.focus(dateField);
    let calendarMenu = screen.getByTestId(
      testId + '-calendar-menu'
    ) as HTMLElement | null;
    expect(calendarMenu).toBeInTheDocument();
    fireEvent.blur(dateField);
    calendarMenu = screen.queryByTestId(
      testId + '-calendar-menu'
    ) as HTMLElement | null;
    expect(calendarMenu).not.toBeInTheDocument();
  });

  // Additonal tests can be added here
});
