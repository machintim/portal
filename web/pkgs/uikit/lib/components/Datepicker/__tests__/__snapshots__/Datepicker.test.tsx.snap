// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Datepicker > Snapshot: date provided 1`] = `
<div
  class="css-1q58hts"
  data-testid="ucl-datepicker"
  tabindex="0"
>
  <div
    aria-label="Date input"
    class="css-1jl4u9h"
    data-testid="ucl-datepicker-date-field"
    role="group"
  >
    <div>
      <input
        aria-label="Date"
        class="css-1gbj3xb"
        data-testid="ucl-datepicker-input"
        inputmode="numeric"
        placeholder="DD/MM/YYYY"
        type="text"
        value="04/02/2025"
      />
    </div>
    <svg
      class="ucl-icon css-148hpxb"
      data-testid="ucl-datepicker-calendar-icon"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="4"
      />
      <line
        x1="16"
        x2="16"
        y1="2"
        y2="6"
      />
      <line
        x1="8"
        x2="8"
        y1="2"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="10"
        y2="10"
      />
    </svg>
  </div>
</div>
`;

exports[`Datepicker > Snapshot: no date provided 1`] = `
<div
  class="css-1q58hts"
  data-testid="ucl-datepicker"
  tabindex="0"
>
  <div
    aria-label="Date input"
    class="css-1jl4u9h"
    data-testid="ucl-datepicker-date-field"
    role="group"
  >
    <div>
      <input
        aria-label="Date"
        class="css-1gbj3xb"
        data-testid="ucl-datepicker-input"
        inputmode="numeric"
        placeholder="DD/MM/YYYY"
        type="text"
        value=""
      />
    </div>
    <svg
      class="ucl-icon css-148hpxb"
      data-testid="ucl-datepicker-calendar-icon"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="4"
      />
      <line
        x1="16"
        x2="16"
        y1="2"
        y2="6"
      />
      <line
        x1="8"
        x2="8"
        y1="2"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="10"
        y2="10"
      />
    </svg>
  </div>
</div>
`;

exports[`Datepicker > Snapshot: with :focus & date provided 1`] = `
<div
  class="css-1q58hts"
  data-testid="abc"
  tabindex="0"
>
  <div
    aria-label="Date input"
    class="css-1jl4u9h"
    data-testid="abc-date-field"
    role="group"
  >
    <div>
      <input
        aria-label="Date"
        class="css-1gbj3xb"
        data-testid="abc-input"
        inputmode="numeric"
        placeholder="DD/MM/YYYY"
        type="text"
        value="04/02/2025"
      />
    </div>
    <svg
      class="ucl-icon css-148hpxb"
      data-testid="abc-calendar-icon"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="18"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="4"
      />
      <line
        x1="16"
        x2="16"
        y1="2"
        y2="6"
      />
      <line
        x1="8"
        x2="8"
        y1="2"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="10"
        y2="10"
      />
    </svg>
  </div>
  <div
    class="css-1adqrcf"
    data-testid="abc-calendar-menu"
  >
    <div
      class="css-ae7n9c"
    >
      <svg
        class="ucl-icon css-stisf5"
        data-testid="ucl-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="15 18 9 12 15 6"
        />
      </svg>
      <span
        class="css-m9j3hv"
      >
        <span>
          February
        </span>
        <span>
          2025
        </span>
      </span>
      <svg
        class="ucl-icon css-stisf5"
        data-testid="ucl-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </div>
    <div
      class="css-10gjhii"
    >
      <div
        class="css-76sg87"
      >
        M
      </div>
      <div
        class="css-76sg87"
      >
        T
      </div>
      <div
        class="css-76sg87"
      >
        W
      </div>
      <div
        class="css-76sg87"
      >
        T
      </div>
      <div
        class="css-76sg87"
      >
        F
      </div>
      <div
        class="css-ie1qp6"
      >
        S
      </div>
      <div
        class="css-ie1qp6"
      >
        S
      </div>
      <div
        aria-label="Select Mon Jan 27 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          27
        </div>
      </div>
      <div
        aria-label="Select Tue Jan 28 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          28
        </div>
      </div>
      <div
        aria-label="Select Wed Jan 29 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          29
        </div>
      </div>
      <div
        aria-label="Select Thu Jan 30 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          30
        </div>
      </div>
      <div
        aria-label="Select Fri Jan 31 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          31
        </div>
      </div>
      <div
        aria-label="Select Sat Feb 01 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          1
        </div>
      </div>
      <div
        aria-label="Select Sun Feb 02 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          2
        </div>
      </div>
      <div
        aria-label="Select Mon Feb 03 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          3
        </div>
      </div>
      <div
        aria-label="Select Tue Feb 04 2025"
        class="css-1rr2g7p"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          4
        </div>
      </div>
      <div
        aria-label="Select Wed Feb 05 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          5
        </div>
      </div>
      <div
        aria-label="Select Thu Feb 06 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          6
        </div>
      </div>
      <div
        aria-label="Select Fri Feb 07 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          7
        </div>
      </div>
      <div
        aria-label="Select Sat Feb 08 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          8
        </div>
      </div>
      <div
        aria-label="Select Sun Feb 09 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          9
        </div>
      </div>
      <div
        aria-label="Select Mon Feb 10 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          10
        </div>
      </div>
      <div
        aria-label="Select Tue Feb 11 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          11
        </div>
      </div>
      <div
        aria-label="Select Wed Feb 12 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          12
        </div>
      </div>
      <div
        aria-label="Select Thu Feb 13 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          13
        </div>
      </div>
      <div
        aria-label="Select Fri Feb 14 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          14
        </div>
      </div>
      <div
        aria-label="Select Sat Feb 15 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          15
        </div>
      </div>
      <div
        aria-label="Select Sun Feb 16 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          16
        </div>
      </div>
      <div
        aria-label="Select Mon Feb 17 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          17
        </div>
      </div>
      <div
        aria-label="Select Tue Feb 18 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          18
        </div>
      </div>
      <div
        aria-label="Select Wed Feb 19 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          19
        </div>
      </div>
      <div
        aria-label="Select Thu Feb 20 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          20
        </div>
      </div>
      <div
        aria-label="Select Fri Feb 21 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          21
        </div>
      </div>
      <div
        aria-label="Select Sat Feb 22 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          22
        </div>
      </div>
      <div
        aria-label="Select Sun Feb 23 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          23
        </div>
      </div>
      <div
        aria-label="Select Mon Feb 24 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          24
        </div>
      </div>
      <div
        aria-label="Select Tue Feb 25 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          25
        </div>
      </div>
      <div
        aria-label="Select Wed Feb 26 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          26
        </div>
      </div>
      <div
        aria-label="Select Thu Feb 27 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          27
        </div>
      </div>
      <div
        aria-label="Select Fri Feb 28 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-1n0xgdm"
        >
          28
        </div>
      </div>
      <div
        aria-label="Select Sat Mar 01 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          1
        </div>
      </div>
      <div
        aria-label="Select Sun Mar 02 2025"
        class="css-1pbj7ec"
        role="button"
      >
        <div
          class="css-6lpfgq"
        >
          2
        </div>
      </div>
    </div>
  </div>
</div>
`;
